projects/core/components/hosted-dropdown/hosted-dropdown.component.ts

```diff
@@ -42,6 +43,17 @@ export interface TuiHostedDropdownContext
     close(): void;
 }
 
+function shouldEscape(this: TuiHostedDropdownComponent, event: KeyboardEvent): boolean {
+    const wrapperElement = this.wrapperElement;
+
+    return (
+        event.key.toLowerCase() === 'escape' &&
+        this.isLastOpenedDropdown &&
+        !!wrapperElement &&
```

Comment:
What's the point of `wrapperElement` and especially checking if it's inside the zone? It **cannot** be outside the zone if it exists. Besides there's an early return for `!this.open` in `onKeyDownEsc` which already handles the case of ignoring escape when dropdown is closed (this early return should probably be moved in `shouldEscape` so that early return is handled in one place).

Reply:
fixed